pkg/reconciler/reconciler.go

```diff
@@ -538,9 +551,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 			updater.EnsureConditionUnknown(conditions.TypeReleaseFailed),
 			updater.EnsureDeployedRelease(nil),
 		)
-		// NOTE: If obj has the uninstall finalizer, that means a release WAS deployed at some point
-		//   in the past, but we don't know if it still is because we don't have an actionClient to check.
-		//   So the question is, what do we do with the finalizer? We could:
```

Comment:
Is this comment line/question still relevant? If so, leave it in. If not, it seems like we need to make further tweaks in the rest of the comment below. Otherwise, the various options we list out now seem somewhat out-of-nowhere.